lib/spack/spack/fetch_strategy.py

```diff
@@ -786,7 +786,9 @@ def git(self):
             # with git as well.
             if not spack.config.get("config:verify_ssl"):
                 self._git.add_default_env("GIT_SSL_NO_VERIFY", "true")
-
+            # stupid fix for mirroring
+            if not spack.config.get("config:git_terminal_prompt_enable"):
+                self._git.add_default_env("GIT_TERMINAL_PROMPT", "0")
```

Comment:
Instead of having a config option for this, could you instead do:

```suggestion
            if not sys.stdout.isatty():
                self._git.add_default_env("GIT_TERMINAL_PROMPT", "0")
```

That will preserve the prompt for interactive sessions, but will disable it in CI, which is where I think you want to avoid the prompts -- yes?

@robertu94: would that address your concern?

Reply:
If we do this my suggestion is that we remove the config option.

Reply:
yes.